### PR TITLE
Update

### PR DESCRIPTION
**Pull request by ThisTestUser**
This includes #128 and #121, so you could close those if you merge this.
This includes a better PeepholeOptimizer, a fake exception remover (Zelix), and an enhanced Smoke deobfuscator.
I also fixed Allatori string decrypt, fixing a lot of issues.

### CODE REVIEW

**Comment by CrackFileso** [comment]
thanks

File: src/main/java/com/javadeobfuscator/deobfuscator/executor/MethodExecutor.java [hunk 1]
```diff
@@ -911,8 +911,9 @@ private static void doFloatMathReturnInteger(List<JavaValue> stack, BiFunction<F
                     case TABLESWITCH: {
                         int x = stack.remove(0).intValue();
                         TableSwitchInsnNode cast = (TableSwitchInsnNode) now;
-                        if (x < cast.labels.size() && x >= 0) {
-                            now = cast.labels.get(x);
+                        int offset = cast.min;
```
Comment (Janmm14) [inline src/main/java/com/javadeobfuscator/deobfuscator/executor/MethodExecutor.java]: I think I messaged this bug to samczsun on spigotmc some months ago.

File: src/main/java/com/javadeobfuscator/deobfuscator/transformers/normalizer/SourceFileClassNormalizer.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    public SourceFileClassNormalizer(Map<String, WrappedClassNode> classes, Map<String, WrappedClassNode> classpath) {
+        super(classes, classpath);
+    }
+
+    @Override
+    public void transform() throws Throwable {
+    	AtomicInteger num = new AtomicInteger(1);
+        CustomRemapper remapper = new CustomRemapper();
+        classNodes().stream().map(WrappedClassNode::getClassNode).forEach(classNode -> {
+            String packageName = classNode.name.lastIndexOf('/') == -1 ? "" : classNode.name.substring(0, classNode.name.lastIndexOf('/') + 1);
```
Comment (Janmm14) [inline src/main/java/com/javadeobfuscator/deobfuscator/transformers/normalizer/SourceFileClassNormalizer.java]: Maybe outsource this boilerplate code somewhere that the class name normalizer and this one is using the same code?

**Comment by samczsun** [comment]
Hey, sorry I've been super inactive. I'm gonna try to start putting some time into this project and we'll see where it goes from there

**Comment by samczsun** [comment]
Looks good. Let me know if you're going to add more changes. If not, I'll merge it as-is

**Comment by ThisTestUser** [comment]
Yep, you can merge it.

**Pull request merged**